packages/graphile-utils/src/makeChangeNullabilityPlugin.ts

```diff
@@ -1,49 +1,121 @@
+  type?: boolean | boolean[];
+  args?: {
+    [argName: string]: boolean | boolean[];
+  };
+}
 
 interface ChangeNullabilityRules {
   [typeName: string]: {
-    [fieldName: string]: boolean;
+    [fieldName: string]: boolean | boolean[] | NullabilityOpts;
   };
 }
 
 export default function makeChangeNullabilityPlugin(
   rules: ChangeNullabilityRules
 ): Plugin {
   return (builder: SchemaBuilder, _options: Options) => {
+    function applyNullability<Type extends GraphQLType | GraphQLInputType>(
+      type: Type,
+      allNullabilities: boolean[]
+    ): Type {
+      const [nullability, ...rest] = allNullabilities;
+      const nullableType = type instanceof GraphQLNonNull ? type.ofType : type;
+      let inner = null;
+      if (nullableType instanceof GraphQLList) {
+        inner = new GraphQLList(applyNullability(nullableType.ofType, rest));
+      } else {
+        if (rest.length > 0) {
+          throw new Error(`Too many booleans passed`);
+        }
+        inner = nullableType;
+      }
+      return nullability
+        ? inner
+        : inner === type
+        ? new GraphQLNonNull(inner)
+        : type; // Optimisation if it's already non-null*/;
+    }
+    function applyTypeNullability<Type extends GraphQLType | GraphQLInputType>(
+      type: Type,
+      shouldBeNullable: boolean | boolean[]
+    ): Type {
+      const allNullabilities: boolean[] =
+        shouldBeNullable instanceof Array
+          ? shouldBeNullable
+          : [shouldBeNullable];
+      return allNullabilities.length > 0
+        ? applyNullability(type, allNullabilities)
+        : type;
+    }
     function changeNullability<
       Field extends GraphQLInputFieldConfig | GraphQLFieldConfig<any, any>
     >(field: Field, build: Build, context: Context<Field>): typeof field {
       const {
         Self,
         scope: { fieldName },
       } = context;
-      const typeRules = rules[Self.name];
+      const typeName = Self.name;
+      const typeRules = rules[typeName];
       if (!typeRules) {
         return field;
       }
       const shouldBeNullable = typeRules[fieldName];
       if (shouldBeNullable == null) {
         return field;
       }
-      const {
-        graphql: { getNullableType, GraphQLNonNull },
-      } = build;
-      const nullableType = getNullableType(field.type);
-      return {
-        ...field,
-        type: shouldBeNullable
-          ? nullableType
-          : nullableType === field.type
-          ? new GraphQLNonNull(field.type)
-          : field.type, // Optimisation if it's already non-null
-      };
+      const shouldApplyNullabilities: NullabilityOpts = {};
+      if (
+        typeof shouldBeNullable === "object" &&
+        !(shouldBeNullable instanceof Array)
+      ) {
+        shouldApplyNullabilities.type = shouldBeNullable.type;
+        shouldApplyNullabilities.args = shouldBeNullable.args;
+      } else {
+        shouldApplyNullabilities.type = shouldBeNullable;
+      }
```

Comment:
Do you need to handle boolean here? Also by doing the `Array.isArray` check first the code is easier to follow:

```suggestion
      if (Array.isArray(shouldBeNullable)) {
        shouldApplyNullabilities.type = shouldBeNullable;
      } else if (typeof shouldBeNullable === 'boolean') {
        shouldApplyNullabilities.type = [shouldBeNullable];
      } else {
        shouldApplyNullabilities.type = shouldBeNullable.type;
        shouldApplyNullabilities.args = shouldBeNullable.args;
      }
```

Reply:
❌ I can't add 
`
      } else if (typeof shouldBeNullable === 'boolean') {
        shouldApplyNullabilities.type = [shouldBeNullable];
      }
`
Because the function apply type nullability is used to apply nullability on typeName.fieldName.args.argName as well.
It the same function used to cover
- typeName.fieldName: boolean | boolean[] (when we're sure it's not nullability opts)
- typeName.fieldName.type: boolean | boolean[]
- typeName.fieldName.args.argName: boolean | boolean[]

Reply:
Sorry, I don't see the problem. The arg type could be a list/etc too, so converting a boolean to a boolean array seems fine?

Reply:
If I move the condition there, It won't be applied for typeName.fieldName.args, because the condition that's change a boolean to a boolean array is applied when passing through the function applyTypeNullability

Reply:
@benjie